### PR TITLE
feat(Header): Reduce nesting

### LINKED ISSUE #462: Reduce nesting of Header

### PR DESCRIPTION
**Pull request by FlyersPh9**
- Reduced nesting to max level 3 in Header by creating separate mixins for sections.
- Renamed classes to include a `.iui-header-` prefix:
	- `.iui-header-divider`
	- `.iui-header-chevron`
	- more listed soon…

Closes #462.

### CODE REVIEW

File: src/header/header.scss [hunk 1]
```diff
@@ -25,118 +24,88 @@
...
-    &,
-    &.iui-slim {
-      .iui-left {
-        .iui-button {
-          .iui-header-button-icon.iui-header-button-icon {
-            @include iui-display-none;
-          }
+@mixin iui-page-header-center {
+  @include iui-page-header-zones;
+  justify-content: center;
```
Comment (veekeys) [inline src/header/header.scss:97]: dont think its needed.. This would center inner content horizontally, which is not the point :?

File: src/header/header.scss [hunk 2]
```diff
@@ -25,118 +24,88 @@
...
-          .iui-header-button-icon + .iui-button-label {
-            margin-left: 0;
-          }
-        }
-      }
-    }
-  }
+@mixin iui-page-header-right {
+  @include iui-page-header-zones;
+  justify-content: flex-end;
```
Comment (veekeys) [inline src/header/header.scss:102]: Same here as with center above

File: src/header/header.scss [hunk 3]
```diff
@@ -190,11 +144,11 @@
...
-        &:not(:focus-visible):not(:hover) {
-          @include themed {
-            background-color: t(iui-color-background-1);
-          }
+      &:not(:focus-visible):not(:hover) {
+        @include themed {
+          background-color: t(iui-color-background-1);
         }
       }
     }
```
Comment (veekeys) [inline src/header/header.scss:154]: Could not comment there, but naming of `iui-header-button-icon`, which is under `iui-header-logo` seems weird..

File: src/header/header.scss [hunk 4]
```diff
@@ -252,20 +204,21 @@
...
-      @media (prefers-reduced-motion: no-preference) {
-        transition: all $iui-speed-fast ease;
-      }
-      @include themed {
-        opacity: t(iui-opacity-3);
-      }
+  .iui-description {
+    height: $iui-baseline * 2;
+    @media (prefers-reduced-motion: no-preference) {
+      transition: all $iui-speed-fast ease;
```
Comment (veekeys) [inline src/header/header.scss:212]: is `all` really needed here?

File: src/header/header.scss [hunk 5]
```diff
@@ -252,20 +204,21 @@
...
+    }
+    @include themed {
+      opacity: t(iui-opacity-3);
     }
   }
 
   > .iui-button-icon {
     flex-shrink: 0;
+    box-sizing: content-box;
```
Comment (veekeys) [inline src/header/header.scss:221]: this is getting overriden

File: src/header/header.scss [hunk 6]
```diff
@@ -274,61 +227,72 @@
 
   .iui-header-button-icon {
```
Comment (veekeys) [inline src/header/header.scss:228]: whats the difference between this and `iui-button-icon` ?

File: src/header/header.scss (hunk 5 above)
Comment (veekeys) [reply src/header/header.scss:221]: oh, by the same value defined below..

File: src/header/header.scss [hunk 7]
```diff
@@ -274,61 +227,72 @@
...
     padding: $iui-xs;
+    box-sizing: content-box;
     @include iui-header-button-icon($iui-icons-default, $iui-border-radius);
     @include themed {
       background-color: t(iui-color-background-3);
       fill: t(iui-icons-color);
     }
   }
 
+  img.iui-header-button-icon {
```
Comment (veekeys) [inline src/header/header.scss:238]: do we have example with img as button icon?

File: src/header/header.scss [hunk 8]
```diff
@@ -274,61 +227,72 @@
...
+    padding: 0;
+    width: auto;
+    height: $iui-icons-default + ($iui-xs * 2);
+  }
+
   &.iui-active {
```
Comment (veekeys) [inline src/header/header.scss:244]: Maybe we could have a single set of active rules?
image

File: src/header/header.scss [hunk 9]
```diff
@@ -274,61 +227,72 @@
...
+        .iui-description {
+          @include iui-display-none;
         }
       }
     }
   }
 }
 
 @mixin iui-header-split-button {
+  max-width: 25vw;
```
Comment (veekeys) [inline src/header/header.scss:280]: do you remember the story behind this value?

File: src/header/header.scss [hunk 10]
```diff
@@ -337,7 +301,7 @@
     @include iui-header-button-active;
 
     // Removing button hovers and backgrounds and adding last child hover
-    .iui-button {
+    .iui-header-breadcrumbs-button {
       &.iui-borderless {
```
Comment (veekeys) [inline src/header/header.scss:305]: dont think we use now iui-borderless anywhere.. all this might be gone now (unless now smsth is broken..)

File: src/header/header.scss [hunk 11]
```diff
@@ -359,7 +323,7 @@
     }
```
Comment (veekeys) [inline src/header/header.scss:323]: and is this last-child needed?

File: src/header/header.scss [hunk 12]
```diff
@@ -359,7 +323,7 @@
...
 
     &.iui-disabled {
-      .iui-button {
+      .iui-header-breadcrumbs-button {
         &:last-child:hover {
           @include themed {
             background-color: transparent;
```
Comment (veekeys) [inline src/header/header.scss:329]: couldnt find the place where this is working too :/

File: src/header/header.scss [hunk 13]
```diff
@@ -403,7 +367,8 @@
```
Comment (veekeys) [inline src/header/header.scss:367]: COuld you double check those rules on hover above? I think now it is working differently than before (maybe better?). Now split button is hoverable when active, before it was not.
And simple buttons have no effect on hover..

File: src/header/header.scss [hunk 14]
```diff
@@ -429,3 +394,28 @@
     }
```
Comment (veekeys) [inline src/header/header.scss:394]: these 2 classes are really confusing... `iui-button-icon` and `iui-header-button-icon`.
Seems disabled still has cursor pointer (in simple buttons, not split, which we dont have example for)

File: src/header/header.scss (hunk 1 above)
Comment (FlyersPh9) [reply src/header/header.scss]: Removed it and visuals didn't change, removed.

File: src/header/header.scss (hunk 4 above)
Comment (FlyersPh9) [reply src/header/header.scss]: Changed it to just `height` & `ease-out` (matching the rest of the header transitions).  Now the animation is a bit smoother than before.

File: src/header/header.scss [hunk 15]
```diff
@@ -359,7 +323,7 @@
...
 
     &.iui-disabled {
-      .iui-button {
+      .iui-header-breadcrumbs-button {
         &:last-child:hover {
           @include themed {
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply src/header/header.scss]: Check it out again, I got it to pass! 🎉

File: src/header/header.scss (hunk 10 above)
Comment (FlyersPh9) [reply src/header/header.scss]: All mentions of `.iui-borderless` have been removed from the header CSS.

File: src/header/header.scss (hunk 13 above)
Comment (FlyersPh9) [reply src/header/header.scss:357]: Check again, my latest push I believe fixed this.

File: src/header/header.scss (hunk 14 above)
Comment (FlyersPh9) [reply src/header/header.scss:384]: Disabled cursor has been fixed.

File: src/header/header.scss [hunk 16]
```diff
@@ -25,159 +25,105 @@
...
-  width: $size;
-  height: $size;
-  border-radius: $border-radius;
-  @media (prefers-reduced-motion: no-preference) {
-    transition: width $iui-speed-fast ease-out, height $iui-speed-fast ease-out, margin $iui-speed-fast ease-out;
-  }
+// #region Header buttons
+@mixin iui-header-button-reset {
+  height: 100%;
+  border-radius: 0 !important;
```
Comment (veekeys) [inline src/header/header.scss:82]: nononononono

Comment (FlyersPh9) [reply src/header/header.scss:82]: Oh right, I meant to leave a comment on this.  This is the only instance where I was running into issues.  I'm sure we can get the `!important` removed, I'm just not sure how yet and could use some advice.

File: src/header/header.scss [hunk 17]
```diff
@@ -25,159 +25,105 @@
...
-
-    &,
-    > nav {
-      overflow: hidden;
-      box-sizing: content-box;
-      margin-bottom: -$iui-xs;
-      padding-bottom: $iui-xs;
-    }
+  &.iui-slim {
+    height: round($iui-baseline * 2.5) + $iui-xxs; // Remove `+ $iui-xxs` when box-sizing: border-box is applied to all elements.
```
Comment (veekeys) [inline src/header/header.scss:29]: Mybe we could apply that bordex-box where needed and remove these 'hacks'?

File: src/header/header.scss [hunk 18]
```diff
@@ -25,159 +25,105 @@
...
+  @media (prefers-reduced-motion: no-preference) {
+    .iui-user-icon {
+      &,
+      * {
+        transition: width $iui-speed-fast ease-out, height $iui-speed-fast ease-out, font-size $iui-speed-fast ease-out;
       }
     }
   }
 }
+// #endregion Selection
```
Comment (veekeys) [inline src/header/header.scss:77]: no region of this.. remove

Comment (veekeys) [reply src/header/header.scss:77]: Or rename to the beggining of the region.. if its needed at all

File: src/header/header.scss [hunk 19]
```diff
@@ -25,159 +25,105 @@
...
+@mixin iui-header-button-reset {
+  height: 100%;
+  border-radius: 0 !important;
 }
 
-@mixin iui-header-buttons {
-  .iui-header-button-icon {
-    @include iui-header-button-icon;
-  }
+@mixin iui-header-brand {
```
Comment (veekeys) [inline src/header/header.scss:85]: Sure about the renaming? I liked logo..

File: src/header/header.scss [hunk 20]
```diff
@@ -25,159 +25,105 @@
...
     a#{&}, // rendered as anchor element
     button#{&}, // rendered as button element
     &[role='button'] {
-      @include iui-header-button-focus;
+      @include iui-focus;
       cursor: pointer;
+      transition: background-color $iui-speed-fast ease-out;
 
       // #region resets for button
       @include iui-reset;
```
Comment (veekeys) [inline src/header/header.scss:129]: could you double check, if all needed? color for sure not

File: src/header/header.scss [hunk 21]
```diff
@@ -25,159 +25,105 @@
...
-  .iui-header-split-button {
-    @include iui-header-split-button;
-  }
-}
+  &-label {
+    margin-right: $iui-sm;
+    font-size: $iui-font-size-leading;
+    transition: color $iui-speed-fast ease-out;
+    @media (prefers-reduced-motion: no-preference) {
+      transition: color $iui-speed-fast ease-out, font-size $iui-speed-fast ease-out;
```
Comment (veekeys) [inline src/header/header.scss:112]: when color is changed?

File: src/header/header.scss [hunk 22]
```diff
@@ -190,181 +136,191 @@
...
-  &:focus:not(:focus-visible) {
-    outline: 0;
-  }
-}
-
-@mixin iui-header-button {
+@mixin iui-header-breadcrumbs-button {
+  @include iui-button;
+  @include iui-button-borderless;
   @include iui-button-size(large, borderless);
```
Comment (veekeys) [inline src/header/header.scss:159]: after all these includes there are so many rules, that its hard to follow.. And there are some which are not used or overriden.. But I guess it would take too long to check one by one and mb write separately. So lets keep them, for now..

File: src/header/header.scss [hunk 23]
```diff
@@ -190,181 +136,191 @@
...
-      }
-      @include themed {
-        opacity: t(iui-opacity-3);
-      }
+    @media (max-width: 768px) {
+      margin-left: 0;
     }
   }
 
   > .iui-button-icon {
```
Comment (veekeys) [inline src/header/header.scss:193]: dont all breadrcumbs buttons have `iui-header-button-icon` class?

Comment (veekeys) [reply src/header/header.scss:193]: a ok.. so its start icon only,, and iui-button-icon is also caret.. hmm..

File: src/header/header.scss [hunk 24]
```diff
@@ -190,181 +136,191 @@
...
+
+    @media (max-width: 768px) {
+      display: none;
+    }
+  }
+
+  img.iui-header-button-icon {
+    padding: 0;
+    width: auto;
+    height: $iui-icons-default + ($iui-xs * 2);
```
Comment (veekeys) [inline src/header/header.scss:220]: `iui-icons-large` ?

File: src/header/header.scss [hunk 25]
```diff
@@ -190,181 +136,191 @@
...
     padding: 0 $iui-xs;
   }
 
   > *:first-child {
-    .iui-button {
+    .iui-header-breadcrumbs-button {
       padding-left: $iui-sm;
+
+      &:hover {
+        background-color: transparent;
```
Comment (veekeys) [inline src/header/header.scss:297]: Is this needed? cause now hover on projects button and imodels button is different by 0.1 opacity :DD

File: src/header/header.scss [hunk 26]
```diff
@@ -190,181 +136,191 @@
...
+    height: $iui-icons-default + ($iui-xs * 2);
   }
 
   &.iui-active {
     @include iui-header-button-active;
   }
 
-  &:disabled {
+  &.iui-disabled,
+  &:disabled,
```
Comment (veekeys) [inline src/header/header.scss:228]: I think hover is broken...
Can we add in demo disabled states too? Same as with split buttons, so we coudl test easier

File: src/header/header.scss [hunk 27]
```diff
@@ -429,3 +386,20 @@
...
+@mixin iui-header-divider {
+  width: 1px;
+  height: $iui-baseline * 2;
+  @include themed {
+    background-color: t(iui-color-background-4);
+  }
+}
+
+@mixin iui-header-chevron {
+  @include iui-icons-small;
```
Comment (veekeys) [inline src/header/header.scss:401]: maybe worth including this mixin for all icons in header? it adds flex display, not sure its gonna be better or worse..

File: src/header/header.scss [hunk 28]
```diff
@@ -25,159 +25,105 @@
...
-      padding-bottom: $iui-xs;
-    }
+  &.iui-slim {
+    height: round($iui-baseline * 2.5) + $iui-xxs; // Remove `+ $iui-xxs` when box-sizing: border-box is applied to all elements.
 
-    .iui-chevron {
-      @include iui-icons-small;
-      flex-shrink: 0;
-      margin: 0 $iui-xs;
+    .iui-user-icon {
```
Comment (veekeys) [inline src/header/header.scss:31]: I guess ideally we could have added class on this one too... ? Not sure if that was better, though

File: src/header/header.scss [hunk 29]
```diff
@@ -190,181 +136,191 @@
...
+      .iui-header-breadcrumbs-button {
+        .iui-header-button-icon {
+          width: $iui-icons-small;
+          height: $iui-icons-small;
+          padding: $iui-xxs;
+          margin: 0 $iui-xs;
+        }
+
+        img.iui-header-button-icon {
+          width: $iui-icons-small + ($iui-xxs * 2);
```
Comment (veekeys) [inline src/header/header.scss:248]: This is inconsistent.. On normal size you set width as `auto`, so maybe it should have width defined in exact number in both cases?

File: src/header/header.scss (hunk 8 above)
Comment (veekeys) [reply src/header/header.scss:223]: Also this one looks weird..
image

File: src/header/header.scss (hunk 16 above)
Comment (veekeys) [reply src/header/header.scss:82]: Can you tell me the exact place, when its broken without it? Not sure I could find it..

File: src/header/header.scss (hunk 17 above)
Comment (FlyersPh9) [reply src/header/header.scss:29]: When I started this PR I actually removed that hack, but it caused test images to change so I left it as is.  I think our original plan was when we add `* {box-sizing: border-box;}` we'd remove the hacky CSS because multiple test images would be changing.

File: backstop/tests/header.html [hunk 30]
```diff
@@ -620,45 +620,45 @@ <h2>
...
-              <span class="iui-label">App name</span>
+              <span class="iui-header-brand-label">App name</span>
             </div>
 
-            <div class="iui-divider"></div>
+            <div class="iui-header-divider"></div>
 
             <!-- Project Button -->
             <nav aria-label="breadcrumbs">
               <span class="iui-button-split-menu iui-header-split-button">
```
Comment (mayank99) [inline backstop/tests/header.html:643]: Now it's a weird situation where we have a mix of our readymade button components and these new header button variants. Will need to work on simplifying this split button probably

File: src/header/header.scss [hunk 31]
```diff
@@ -190,181 +136,191 @@
...
   }
 
   > * + * {
     margin-left: $iui-sm;
   }
 
   .iui-header-button-icon {
+    display: inline-flex;
     padding: $iui-xs;
+    box-sizing: content-box;
```
Comment (mayank99) [inline src/header/header.scss:205]: What issue does this fix and how? Feels hacky and we shouldn't rely on it.

File: src/header/header.scss [hunk 32]
```diff
@@ -429,3 +386,20 @@
     }
   }
 }
+// #endregion Header buttons
+
+// #region Header extras
+@mixin iui-header-divider {
+  width: 1px;
+  height: $iui-baseline * 2;
```
Comment (mayank99) [inline src/header/header.scss:394]: I was playing with dividers in the integration sprint and realized that `width: 1px` is not reliable on high resolution screens. Try using borders instead https://ishadeed.com/article/flexbox-separator/

File: src/header/header.scss [hunk 33]
```diff
@@ -190,181 +136,191 @@
...
     }
 
-    &.iui-disabled {
-      .iui-button {
-        &:last-child:hover {
-          @include themed {
-            background-color: transparent;
-          }
-        }
+    > *:last-child > .iui-header-breadcrumbs-button:hover {
```
Comment (mayank99) [inline src/header/header.scss:321]: 0-5-0 🤔

File: src/header/header.scss [hunk 34]
```diff
@@ -190,181 +136,191 @@
...
-          }
+        .iui-header-breadcrumbs-button-description {
+          @include iui-display-none;
         }
       }
     }
   }
 }
 
+@mixin iui-header-button {
```
Comment (mayank99) [inline src/header/header.scss:261]: This is getting confusing. Why is there a `header-button` and a `header-breadcrumbs-button`?

The example in your description doesn't apply here because those extra buttons are passed by the user.

File: src/header/header.scss (hunk 17 above)
Comment (mayank99) [reply src/header/header.scss:29]: I kinda like that no images changed in this PR
image

File: src/header/header.scss (hunk 5 above)
Comment (FlyersPh9) [reply src/header/header.scss:195]: You are correct, removed from here & left on `.iui-header-button-icon`.

File: src/header/header.scss (hunk 6 above)
Comment (FlyersPh9) [reply src/header/header.scss:202]: `.iui-header-button-icon` is the icon that appears within our breadcrumb buttons.  `.iui-button-icon` applies to all of the icons within buttons in the header (breadcrumb button icons as well as notification, help, etc.).

File: src/header/header.scss (hunk 7 above)
Comment (FlyersPh9) [reply src/header/header.scss:217]: We do not currently but I could add one.  I did just now test it locally and found 2 issues, so we probably should add a test for this.

Comment (FlyersPh9) [reply src/header/header.scss:217]: Also worth pointing out that our react header is using `` & these changes will fix the display bug in react:

**React**
Screen Shot 2022-02-18 at 4 17 13 PM

**CSS**
Screen Shot 2022-02-18 at 4 17 01 PM

File: src/header/header.scss (hunk 9 above)
Comment (FlyersPh9) [reply src/header/header.scss:278]: I think we needed to set a width for truncation to take place and we landed on 25% viewport width. 🤷🏻‍♂️

File: src/header/header.scss (hunk 34 above)
Comment (FlyersPh9) [reply src/header/header.scss:261]: We can support placing 3 types of buttons in the header with unique styling:  a breadcrumb button, a header button, and a regular button (maybe 4 if you count the branding button as well).  Our previous implementation if a user passes a regular button they would get this:
Screen Shot 2022-02-18 at 4 25 37 PM

This `.iui-header-button` modifier ensures that a user passed regular button's styling doesn't get overriden.

Comment (mayank99) [reply src/header/header.scss:261]: We cannot apply any modifiers on user's buttons, because we don't know what element they will pass.

File: src/header/header.scss (hunk 31 above)
Comment (FlyersPh9) [reply src/header/header.scss:205]: In iTwinUI-Layouts Vilius is applying `* { box-sizing: border-box; }` & I know we plan on doing that across all `iui-` classes eventually.  In our layout, we are seeing this:
Screen Shot 2022-02-18 at 4 31 43 PM
 
It should look like this & `content-box` fixes it:
Screen Shot 2022-02-18 at 4 32 35 PM

File: src/header/header.scss [hunk 35]
```diff
@@ -5,9 +5,9 @@
 @import '../icon/index';
 @import '../button/index';
 
+// #region Header structure
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply src/header/header.scss:8]: I've found them helpful for collapsing sections of code when working on long files.  If you really want them removed, let me leave them until the very end and then I can remove them.
Screen Shot 2022-02-18 at 4 35 02 PM

File: src/header/classes.scss [hunk 36]
```diff
@@ -6,6 +6,42 @@
...
+
+.iui-page-header-center {
+  @include iui-page-header-center;
+}
+
+.iui-page-header-right {
+  @include iui-page-header-right;
+}
+
+.iui-header-brand {
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply src/header/classes.scss:21]: Our current implementation has a blend of both as well.  I tried to keep on trend with that by deciding `iui-page-header` has to do with the header bar and the zones within it, & `iui-header` has to do with the content inside of `iui-page-header`.  I am ok with changing to make sure all align, but which should I use?  I'm pretty sure originally we used `iui-header` but I think it was changed along the way.

File: src/header/header.scss (hunk 25 above)
Comment (FlyersPh9) [reply src/header/header.scss:297]: If I remove this line we get a double hover effect on the left part of a split button:
Screen Recording 2022-02-18 at 4 43 40 PM

File: src/header/header.scss (hunk 24 above)
Comment (FlyersPh9) [reply src/header/header.scss:220]: Replaced the math with a singular icon variable in both instances of `img.iui-header-button-icon`.

File: src/header/header.scss (hunk 21 above)
Comment (FlyersPh9) [reply src/header/header.scss:112]: Really seems to only be when switching themes.  Removing color transition here and fill transition for the brand icon for the same reason (although it is visually inconsistent with the other buttons in the header).

File: src/header/header.scss (hunk 18 above)
Comment (FlyersPh9) [reply src/header/header.scss]: Renamed.

File: src/header/header.scss [hunk 37]
```diff
@@ -25,159 +25,105 @@
...
   }
+}
 
-  > .iui-center {
-    justify-content: center;
-  }
-
-  > .iui-right {
-    justify-content: flex-end;
+@mixin iui-page-header-zones {
```
[the remark this answers is not in the document]
Comment (mayank99) [reply src/header/header.scss]: These can be combined under `&, nav`
image

File: src/header/header.scss (hunk 3 above)
Comment (FlyersPh9) [reply src/header/header.scss:137]: Tried to cleanup the mixin & class names.  We now have:  `.iui-header-brand-icon`, `.iui-header-breadcrumbs-button-icon`, & `.iui-button-icon`.

File: src/header/header.scss (hunk 26 above)
Comment (FlyersPh9) [reply src/header/header.scss:220]: Just saw that now too, appears the right side of a disabled split button is broken.  Looking into it.

File: src/header/header.scss (hunk 16 above)
Comment (veekeys) [reply src/header/header.scss:82]: I removed important and it seemed to still work fine as the only rule with border radius comes from buttons mixin, which is above..

File: src/header/header.scss (hunk 25 above)
Comment (veekeys) [reply src/header/header.scss]: But isnt it how it works now??

File: src/header/header.scss (hunk 19 above)
Comment (adamhock) [reply src/header/header.scss:85]: I like the renaming because `iui-header-brand` includes both the `icon` and `label`. Having a class name `iui-header-logo-icon` sounds redundant in my opinion

File: backstop/tests/menu.html [hunk 38]
```diff
@@ -493,7 +493,7 @@ <h3>Header</h3>
               ></svg-model>
               <span class="iui-button-label">
                 <div>Model Beta</div>
```
Comment (adamhock) [inline backstop/tests/menu.html:495]: Does this need to be changed to have a class name of `iui-header-breadcrumbs-button-name` like the Project Button example?

**Review by adamhock — CHANGES_REQUESTED** [review]

File: backstop/tests/menu.html [hunk 39]
```diff
@@ -629,7 +629,7 @@ <h3>Header</h3>
...
```
Comment (adamhock) [inline backstop/tests/menu.html:631]: Does this need to be changed to have a class name of `iui-header-breadcrumbs-button-name` like the Project Button example?

File: src/header/header.scss (hunk 22 above)
Comment (FlyersPh9) [reply src/header/header.scss]: Big rewrite, most if not all `@includes` have been removed.

File: src/header/header.scss (hunk 25 above)
Comment (FlyersPh9) [reply src/header/header.scss]: That is not how it is now.  In our current implementation if you hover over the left part of the split button you do NOT get a double hover effect on the left portion.  See this GIF and compare Project Alpha's hover states:
Screen Recording 2022-03-01 at 4 16 46 PM

File: src/header/header.scss (hunk 26 above)
Comment (mayank99) [reply src/header/header.scss]: Hover is still broken
image

I've added the disabled toggle to all sections in [`75c937e`](https://github.com/iTwin/iTwinUI/pull/463/commits/75c937e2f1b510876ec5a3fde6311ef1433fd7af).

File: backstop/tests/menu.html (hunk 39 above)
Comment (FlyersPh9) [reply backstop/tests/menu.html]: Yeah, I think it does.  It should be updated in the latest push.

File: src/header/header.scss (hunk 34 above)
Comment (FlyersPh9) [reply src/header/header.scss]: Ok, this change has been reverted.  All `.iui-button` within the right section will be forced 100% height, but other sections are uneffected (which allows the search button to work in the center section).

File: src/header/header.scss (hunk 33 above)
Comment (FlyersPh9) [reply src/header/header.scss]: I was able to reduce specificity of this by applying a class directly.

File: src/header/header.scss (hunk 35 above)
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply src/header/header.scss]: Regions have been removed & mixins have been separated into 2 different .scss files.

File: src/header/classes.scss (hunk 36 above)
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply src/header/classes.scss]: Everything within `header.scss` is using `.iui-page-header-` and then everything within `header-buttons.scss` is using `.iui-header-`.

File: src/header/header.scss (hunk 26 above)
Comment (FlyersPh9) [reply src/header/header.scss]: Hover is now fixed.

File: src/header/header-buttons.scss [hunk 40]
```diff
@@ -0,0 +1,346 @@
...
+    cursor: not-allowed;
+
+    svg {
+      @include themed {
+        fill: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-5));
+      }
+    }
+  }
+
+  @at-root {
```
Comment (mayank99) [inline src/header/header-buttons.scss:209]: The rules inside this `@at-root` don't belong in this mixin. If you look at the compiled output, you will find that none of these have the `iui-header-breadcrumb-button` class (which is the name of the mixin these are rules in currently)

File: src/header/header-buttons.scss [hunk 41]
```diff
@@ -0,0 +1,346 @@
...
+  will-change: transform;
+  @media (prefers-reduced-motion: no-preference) {
+    transition: transform $iui-speed-fast ease-out;
+  }
+
+  @at-root {
+    .iui-disabled,
+    :disabled,
+    [aria-disabled='true'] {
+      .iui-header-breadcrumb-button-icon {
```
Comment (mayank99) [inline src/header/header-buttons.scss:341]: If you remove nesting from the place that this mixin is included, then you can just use `&` instead of this line.

Every time you type out the name of a class, you should feel some "tingling" 😄

File: src/header/header-buttons.scss [hunk 42]
```diff
@@ -0,0 +1,346 @@
...
+      }
+
+      .iui-header-breadcrumb-button-dropdown:hover {
+        @include themed {
+          background-color: rgba(t(iui-color-foreground-primary-rgb), t(iui-opacity-6));
+        }
+      }
+
+      // Blue text
+      .iui-button-label {
```
Comment (mayank99) [inline src/header/header-buttons.scss:228]: are we using `iui-button-label` anywhere?

File: backstop/tests/header.html [hunk 43]
```diff
@@ -104,86 +112,122 @@ <h1>Header</h1>
...
+                    class="iui-header-breadcrumb-button-icon"
+                    src="../assets/image-test.jpg"
+                    draggable="false"
+                  />
+                  <span class="iui-header-breadcrumb-button-label">
+                    <div class="iui-header-breadcrumb-button-label-name">Project Alpha</div>
+                    <div class="iui-header-breadcrumb-button-label-description">0x0123456789</div>
+                  </span>
+                  <svg-caret-down-small
+                    class="iui-button-icon"
```
Comment (mayank99) [inline backstop/tests/header.html:169]: This one feels random. Could we avoid this `iui-button-icon` class if we are not using `iui-button` anywhere?

File: src/header/header.scss (hunk 37 above)
[the remark this answers is not in the document]
Comment (mayank99) [reply src/header/header.scss]: this is what I'm suggesting
image

File: backstop/tests/menu.html (hunk 39 above)
Comment (adamhock) [reply backstop/tests/menu.html]: I don't see the update? I thought these cases would be changed to `<div class="iui-header-breadcrumb-button-label-name">Model Beta</div>`

**Comment by FlyersPh9** [comment]
Will be submitting a new PR after #568 passes.
